src/utils/annotation.ts

```diff
@@ -199,3 +199,14 @@ export function tagFilterFunction(
   }
   return filterTags.every((filterTag) => inputTags.includes(filterTag));
 }
+
+export function tagCloudFilterFunction(
+  inputTags: string[],
+  filterTags: string[],
+  exclusive: boolean,
+) {
+  if (exclusive && inputTags.length !== filterTags.length) {
+    return false;
+  }
+  return inputTags.some((inputTag) => filterTags.includes(inputTag));
+}
```

Comment:
I don't understand what this function should do.
For example, if you you have:
- `inputTags = ["x", "y", "z"]`
- `filterTags = ["a", "b", "c", "x"]`
It will return `true` when `exclusive` is `false` and return `false` when `exclusive` is `true`.

Also, the logic for tags should be consistent through all the application, so why not changing the logic of `tagFilterFunction` instead of creating a new function (`tagCloudFilterFunction`)?

Reply:
Good question! What I found is that people wanted to be able to basically have a little "pad" on which they could click to turn on and off various annotations (mostly, people use a single tag for annotations). This logic allows them to easily turn them on and off. When exclusive is false, it runs in basically an "or" mode, in which selecting a particular tag will show every annotation that has any of those tags. When exclusive is true, it runs like a real "exclusive" mode, in which show only objects that are an exact match for the selected tags. Honestly, not many/any people use the exclusive mode anyway, but I felt like there may be moments when someone wants to very specifically find a set of annotations, so I left that in. TL;DR: I came up with logic that matches user expectations with a feature many people had been requesting (and that I had noticed people outside the lab were looking for when using it).

Reply:
Then I think we should not create a new function, but apply this logic to all the codebase.
The logic should work as you expect when `exclusive` is `false`, but when `exclusive` is `true`, it doesn't really do what you expect
Do you want me to fix it?

Reply:
I thought about that a lot, and I think the logic for the tag cloud is actually quite different than what is expected in the tag picker, which is why I made a separate function.